### PR TITLE
Add extractor for sponsees from an email message

### PR DESCRIPTION
**Pull request by adrianclay**
To sponsor guests for GovWifi you send an email with their details to sponsor@wifi.service.gov.uk

Each email/phone number appears on a new line.

The regression tests have been pulled over from the PHP test suite.

### CODE REVIEW

File: lib/email_sponsees_extractor.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require 'nokogiri'
+
+class EmailSponseesExtractor
+  def execute(s)
```
Comment (adrianclay) [inline lib/email_sponsees_extractor.rb]: Ugg that variable name

File: lib/email_sponsees_extractor.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      lines_from_html(mail)
+    end
+  end
+
+  def lines_from_plain_text(message)
+    message.decoded.lines "\n"
+  end
+
+  def lines_from_html(mail)
+    Nokogiri::HTML(mail.html_part.decoded).xpath("//text()").reject(&:cdata?).map do |node|
```
Comment (chrisblackburn) [inline lib/email_sponsees_extractor.rb]: In the absence of alternative solutions, I'll trust that this is the cleanest way to achieve this :-)

Comment (adrianclay) [reply lib/email_sponsees_extractor.rb]: I've moved the style tag check into the xpath, and it looks nicer I'd say.  Thoughts?

File: Gemfile [hunk 3]
```diff
@@ -1,8 +1,9 @@
 source 'http://rubygems.org'
 ruby File.read('.ruby-version').chomp
 
-gem "mail", "~> 2.7"
+gem 'mail', "~> 2.7"
```
Comment (chrisblackburn) [inline Gemfile]: Don't forget the other double quotes

File: Gemfile [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
 gem 'mysql2'
+gem 'nokogiri', "~> 1.8"
```
Comment (chrisblackburn) [inline Gemfile]: Double quotes here also

**Pull request merged**